method/link.js

```diff
@@ -84,6 +84,50 @@ module.exports = function Link (ssb) {
         }),
         pull.collect(cb)
       )
+    },
+    findSubgroupByGroupId (groupId, cb) {
+      if (!isGroup(groupId)) return cb(new Error(`findSubgroupByGroupId expected a groupId, got ${groupId} instead.`))
+
+      const query = [{
+        $filter: {
+          value: {
+            content: {
+              type: 'link/group-subgroup',
+              parent: groupId,
+              tangles: {
+                link: { root: null, previous: null }
+              }
+            }
+          }
+        }
+      }]
+
+      pull(
+        ssb.query.read({ query }),
+        pull.filter(groupSubgroupLink.spec.isRoot),
+        // commented this out as its making it fail!
+        // pull.filter(link => {
+        //   // return link.value.content.child === link.value.content.recps[0]
+        //   // it would be very strange for a link to be created like this
+        //   // but we should consider it unsafe and ignore it I think
+        // }),
```

Comment:
I want a uniqueness filter here too - don't want the same group linked multiple times

check out pull.unique - http://pull-stream.github.io/#unique